README.md

```diff
@@ -8,15 +8,174 @@
+    assistants:CreateAssistantRequest request = {
+        model: "gpt-3.5-turbo",
+        name: "Math Tutor",
+        description: "An Assistant for personal math tutoring",
+        instructions: "You are a personal math tutor. Help the user with their math questions.",
+        tools: [tool]
+    };
+
+    // call the `post assistants` resource to create an Assistant
+    assistants:AssistantObject assistantResponse = check openAIAssistant->/assistants.post(request, headers);
+}
+```
+
+#### Create a thread
+
+A Thread represents a conversation between a user and one or many Assistants. You can create a Thread when a user (or your AI application) starts a conversation with your Assistant.
+
+```ballerina
+public function main() returns error?{
+    // define the thread request 
+    assistants:CreateThreadRequest createThreadReq = {
+        messages: []
+    };
+
+    // call the `post threads` resource to create a Thread
+    assistants:ThreadObject threadResponse = check openAIAssistant->/threads.post(createThreadReq, headers);
+}
+```
+
+#### Add a message to the thread
+
+The contents of the messages your users or applications create are added as Message objects to the Thread. Messages can contain both text and files. There is no limit to the number of Messages you can add to Threads — the context that does not fit into the model's context window will be truncated automatically.
+
+```ballerina
+public function main() returns error?{
+    string threadId = "your_thread_id";
+
+    // define the message object
+    assistants:CreateMessageRequest createMsgReq = {
+        role: "user",
+        content: "Can you help me solve the equation `3x + 11 = 14`?",
+        metadata: {}
+    };
+
+    // create a message in the thread
+    assistants:MessageObject messageResponse = check openAIAssistant->/threads/[threadId]/messages.post(createMsgReq, headers);
+}
+```
+
+#### Create a run
+
+Once all the user Messages have been added to the Thread, you can Run the Thread with any Assistant. Creating a Run uses the model and tools associated with the Assistant to generate a response. These responses are added to the Thread as Assistant Messages.
+
+```ballerina
+public function main() returns error?{
+    string threadId = "your_thread_id";
+
+    // define the run request object
+    assistants:CreateRunRequest runReq = {
+        assistant_id: "your_assistant_id",
+        model: "gpt-3.5-turbo",
+        instructions: "You are a personal math tutor. Assist the user with their math questions.",
+        temperature: 0.7,
+        top_p: 0.9,
+        max_prompt_tokens: 400,
+        max_completion_tokens: 200
+    };
+
+    // create a run in the thread
+    assistants:RunObject runResponse = check openAIAssistant->/threads/[threadId]/runs.post(runReq, headers);
+
+}
+```
+Once the Run completes, you can list the Messages added to the Thread by the Assistant.
+
+```ballerina
+public function main() returns error?{
```

Comment:
Formatting error. Please re-check all the Ballerina sources in the quickstart guide for formatting errors
```suggestion
public function main() returns error? {
```